src/viam/sdk/components/movement_sensor/movement_sensor.hpp

```diff
@@ -0,0 +1,164 @@
+/// @file components/movementsensor/movementsensor.hpp
+///
+/// @brief Defines a `MovementSensor` component.
+#pragma once
+
+#include <string>
+
+#include <viam/api/component/movementsensor/v1/movementsensor.pb.h>
+
+#include <viam/sdk/common/linear_algebra.hpp>
+#include <viam/sdk/common/proto_type.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/components/movement_sensor/server.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/registry/registry.hpp>
+#include <viam/sdk/resource/resource_manager.hpp>
+#include <viam/sdk/spatialmath/orientation.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup MovementSensor Classes related to the MovementSensor component.
+
+/// @class MovementSensor movementsensor.hpp "components/movementsensor/movementsensor.hpp"
+/// @ingroup MovementSensor
+///
+/// This acts as an abstract base class to be inherited from by any drivers representing
+/// specific movementsensor implementations. This class cannot be used on its own.
+class MovementSensor : public Component {
+   public:
+    /// @struct compassheading
+    struct compassheading {
+        /// A number from 0-359 where 0 is North, 90 is East, 180 is South, and 270 is   West
+        double value;
+    };
```

Comment:
Not hyped that this lives on the movement sensor object, but probably fine

Reply:
want me to change to typedef?
```suggestion
    /// A number from 0-359 where 0 is North, 90 is East, 180 is South, and 270 is West
    typedef double compassheading;
```

Reply:
I don't have a  strong opinion. typedef is probably better but either is fine